src/Interpreters/Access/InterpreterGrantQuery.cpp

```diff
@@ -178,6 +178,21 @@ namespace
                 elements_to_revoke.emplace_back(std::move(element_to_revoke));
         }
 
+        /// Additional check for REVOKE
```

Comment:
It feels logically more right for this code to appear a bit earlier - before we calculate `elements_to_revoke`.